### PR TITLE
Feat/15951/add button secondary

### LINKED ISSUE #15092: [Ext Nav] Create component: ButtonSecondary
### Description
Create a reusable UI component for ButtonSecondary

### References
[Figma Link](https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/Design-System-2.0?node-id=1287%3A11481)
UI component guidelines: TBC
CSS guidelines: TBC
Testing guidelines: TBC

### Files needed
- `index.js`
- `index.scss`
- `ButtonSecondary.js`
- `ButtonSecondary.stories.js`
- `ButtonSecondary.test.js`
- `README.mdx`

### Technical details
TBC

### Acceptance criteria
- [ ] Uses color, typography, shadows design tokens 
- [ ] Uses semantic html
- [ ] PropTypes have descriptions
- [ ] Has storybook story with controls
- [ ] Has documentation in MDX
- [ ] Has unit tests and 90% coverage
- [ ] Works in Chrome and Firefox
- [ ] Performance tested: no unnecessary re-renders or other performance concerns

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation

Adding ButtonSecondary built off of ButtonBase. 

(similar to ButtonPrimary)

Sizes: SM, MD, & LG
Types: Normal (default) and Danger (enabled with danger prop = true) 

[Figma](https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/DS-Components?node-id=2939%3A20593)

## More Information

- Fixes #15092 

## Screenshots/Screencaps

Screen Shot 2022-10-05 at 11 51 24 AM
Screen Shot 2022-10-05 at 12 02 27 PM

## Manual Testing Steps

`yarn test:unit:jest .ui/components/component-library/button-secondary/button-secondary.test.js`

## Pre-Merge Checklist

- [x] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [x] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

### CODE REVIEW

File: ui/components/component-library/button-secondary/README.mdx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import { Story, Canvas, ArgsTable } from '@storybook/addon-docs';
+
+import { ButtonSecondary } from './button-secondary';
+
+# ButtonSecondary
+
+The `ButtonSecondary` is an extension of `ButtonBase` to support primary styles.
```
Comment (georgewrmarshall) [inline ui/components/component-library/button-secondary/README.mdx:7]: Update copy
```suggestion
The `ButtonSecondary` is an extension of `ButtonBase` to support secondary styles.
```

File: ui/components/component-library/button-secondary/button-secondary.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import classnames from 'classnames';
+
+import { ButtonBase } from '../button-base';
+import { BUTTON_SECONDARY_SIZES } from './button-secondary.constants';
+
+export const ButtonSecondary = ({
+  className,
+  danger,
+  size = BUTTON_SECONDARY_SIZES.MD,
+
```
Comment (georgewrmarshall) [inline ui/components/component-library/button-secondary/button-secondary.js:12]: nit: remove line break
```suggestion
```

File: ui/components/component-library/button-secondary/button-secondary.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+   */
+  className: PropTypes.string,
+  /**
+   * Boolean to change button type to Danger when true
+   */
+  danger: PropTypes.bool,
+  /**
+   * The possible size values for ButtonSecondary: 'BUTTON_SECONDARY_SIZES.SM', 'BUTTON_SECONDARY_SIZES.MD', 'BUTTON_SECONDARY_SIZES.LG',
+   * Default value is 'BUTTON_SECONDARY_SIZES.MD'.
+   */
```
Comment (georgewrmarshall) [inline ui/components/component-library/button-secondary/button-secondary.js:38]: I think we should use global `SIZES` const here
```suggestion
   * The possible size values for ButtonSecondary: 'SIZES.SM', 'SIZES.MD', 'SIZES.LG',
   * Default value is 'SIZES.MD'.
   */
```

**Review by georgewrmarshall — COMMENTED** [review]
Some tiny suggestions otherwise 🔥🔥🔥

File: ui/components/component-library/button-secondary/README.mdx (hunk 1 above)
Comment (georgewrmarshall) [reply ui/components/component-library/button-secondary/README.mdx:7]: @garrettbear still need to complete this update?

File: ui/components/component-library/button-secondary/button-secondary.stories.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  },
+  args: {
+    children: 'Button Secondary',
+  },
+};
+
+export const DefaultStory = (args) => (
+  <>
+    <ButtonSecondary {...args} />
+  </>
```
Comment (NidhiKJha) [inline ui/components/component-library/button-secondary/button-secondary.stories.js:110]: Nit: Do we need a fragment here?

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
LGTM! 🔥🔥

**Review by georgewrmarshall — DISMISSED** [review]
LGTM 🔥

**Review by georgewrmarshall — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**